python/ray/train/_internal/logging.py

```diff
@@ -0,0 +1,241 @@
+import logging
+import logging.config
+import os
+from enum import Enum
+from typing import Optional
+
+import yaml
+
+import ray
+from ray.train._internal.session import _get_session
+from ray.train.constants import LOG_CONFIG_PATH_ENV, LOG_ENCODING_ENV
+
+# JSON Encoding format for Ray Train structured logging
+DEFAULT_JSON_LOG_ENCODING_FORMAT = "JSON"
+
+# Default logging configuration for Ray Train
+DEFAULT_LOG_CONFIG_JSON_STRING = {
+    "version": 1,
+    "disable_existing_loggers": False,
+    "formatters": {
+        "ray": {
+            "format": "%(asctime)s\t%(levelname)s %(filename)s:%(lineno)s -- %(message)s"  # noqa: E501
+        },
+        "ray_json": {"class": "ray._private.ray_logging.formatters.JSONFormatter"},
+    },
+    "filters": {
+        "console_filter": {"()": "ray.train._internal.logging.HiddenRecordFilter"},
+        "core_context_filter": {
+            "()": "ray._private.ray_logging.filters.CoreContextFilter"
+        },
+        "train_context_filter": {
+            "()": "ray.train._internal.logging.TrainContextFilter"
+        },
+    },
+    "handlers": {
+        "console_json": {
+            "class": "logging.StreamHandler",
+            "formatter": "ray_json",
+            "filters": ["core_context_filter", "train_context_filter"],
+        },
+        "console_text": {
+            "class": "ray._private.log.PlainRayHandler",
+            "formatter": "ray",
+            "level": "INFO",
+            "filters": ["train_context_filter", "console_filter"],
+        },
+        "file": {
+            "class": "ray.train._internal.logging.SessionFileHandler",
+            "formatter": "ray_json",
+            "filename": "ray-train.log",
+            "filters": ["core_context_filter", "train_context_filter"],
+        },
+    },
+    "loggers": {
+        "ray.train": {
+            "level": "DEBUG",
+            "handlers": ["file", "console_text"],
+            "propagate": False,
+        },
+    },
+}
+
+
+class TrainLogKey(str, Enum):
+    RUN_ID = "run_id"
+    WORLD_SIZE = "world_size"
+    WORLD_RANK = "world_rank"
+    LOCAL_WORLD_SIZE = "local_world_size"
+    LOCAL_RANK = "local_rank"
+    NODE_RANK = "node_rank"
+    # This key is used to hide the log record if the value is True.
+    # By default, train workers that are not ranked zero will hide
+    # the log record.
+    HIDE = "hide"
```

Comment:
I don't think `hide=True` is necessary. At least, the product won't utilize this field.

Reply:
Got it! A followup question: what's the format we should follow so that the product team can use to filter for logs to be shown /hidden by default in the log viewer?

Reply:
we can just filter out non "rank 0" or do whatever behavior based on the other fields